manifests/root_user.pp

```diff
@@ -10,10 +10,14 @@
 # @param manage_group
 #  Ensure the root group has appropriate UIDs, etc
 #
+# @param password
+#  Set the root user's password using Puppet
+#
 class simp::root_user (
-  Boolean $manage_perms = true,
-  Boolean $manage_user  = true,
-  Boolean $manage_group = true
+  Boolean                                         $manage_perms = true,
+  Boolean                                         $manage_user  = true,
+  Boolean                                         $manage_group = true,
+  Variant[Undef, String, Pattern[/^\$\d[ay]?\$/]] $password     = undef,
```

Comment:
This should be something like `Optional[Simplib::ShadowPass]` where `Simplib::ShadowPass` is a custom data type added to `simplib` (or this module if you don't feel like PR'ing another one) that denotes the valid data types.

Reply:
Yeah, I was thinking this as I was doing it...I should've just done the custom type earlier :-P I'll make this happen.